# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmSerialize.cpp
# Commit: 9111720170a2
# Full Hash: 9111720170a283e7c3801856d5c113a2f17caf20
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2022-05-06 09:50:52
# Regressor Bug: 1642412
# File Overlap Count: 7
# Description:
#   Bug 1762171 - wasm: Fix cranelift build for table optimizations. r=yury
#   
#   This commit adds back the builtins that were removed in
#   bug 1642412.
#   
# ==============================================================================

diff -r 2340f31f2559 -r 9111720170a2 js/src/wasm/WasmSerialize.cpp
--- a/js/src/wasm/WasmSerialize.cpp	Thu May 05 21:49:46 2022 +0000
+++ b/js/src/wasm/WasmSerialize.cpp	Thu May 05 21:49:46 2022 +0000
@@ -708,7 +708,7 @@
 CoderResult CodeSymbolicLinkArray(
     Coder<mode>& coder,
     CoderArg<mode, wasm::LinkData::SymbolicLinkArray> item) {
-  WASM_VERIFY_SERIALIZATION_FOR_SIZE(wasm::LinkData::SymbolicLinkArray, 6912);
+  WASM_VERIFY_SERIALIZATION_FOR_SIZE(wasm::LinkData::SymbolicLinkArray, 6984);
   for (SymbolicAddress address :
        mozilla::MakeEnumeratedRange(SymbolicAddress::Limit)) {
     MOZ_TRY(CodePodVector(coder, &(*item)[address]));
@@ -719,7 +719,7 @@
 template <CoderMode mode>
 CoderResult CodeLinkData(Coder<mode>& coder,
                          CoderArg<mode, wasm::LinkData> item) {
-  WASM_VERIFY_SERIALIZATION_FOR_SIZE(wasm::LinkData, 6960);
+  WASM_VERIFY_SERIALIZATION_FOR_SIZE(wasm::LinkData, 7032);
   if constexpr (mode == MODE_ENCODE) {
     MOZ_ASSERT(item->tier == Tier::Serialized);
   }
